### PR TITLE
Subtraction primitive for phylanx

### PR DESCRIPTION
**Pull request by justwagle**
This pull request is a first attempt at creating a subtraction primitive for Phylanx.

### CODE REVIEW

File: phylanx/execution_tree/primitives/sub_operation.hpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                ir::node_data<double> sub2d(operands_type const& ops) const;
+
+                ir::node_data<double> sub1d1d(operands_type const& ops) const;
+                ir::node_data<double> sub2d2d(operands_type const& ops) const;
+
+            private:
+                std::vector<primitive> operands_;
+            };
+        }}}
+
```
Comment (hkaiser) [inline phylanx/execution_tree/primitives/sub_operation.hpp]: What happened to the indentation here?

File: src/execution_tree/primitives/sub_operation.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                }
+            }),
+            detail::map(operands_,
+                [](primitive const& p) -> hpx::future<ir::node_data<double>>
+                {
+                    return p.eval();
+                })
+        );
+    }
+}}}
```
Comment (hkaiser) [inline src/execution_tree/primitives/sub_operation.cpp]: This implementation will support working with exectly 2 operands only. Do you think you could extend it to work with an arbitrary amount of operands (>= 2)?

Comment (justwagle) [reply src/execution_tree/primitives/sub_operation.cpp]: Yes, I just pushed a version that should be able support more than 2 versions in e8f78115de03af2ca03566ebeb1ca4e454285503.

File: phylanx/execution_tree/primitives/sub_operation.hpp (hunk 1 above)
Comment (justwagle) [reply phylanx/execution_tree/primitives/sub_operation.hpp]: It was CLion wanting to change formatting automatically :( This is fixed with db121c1c1bcb974caf19314a2865ec5ec6038dc0

**Comment by justwagle** [comment]
Added test cases to generate_tree.cpp in  b22f81062393e1fef6bed3f8e5418afe4f168c00

**Comment by hkaiser** [comment]
@justwagle Could you resolve the conflicts, please? Otherwise this can't be merged.

**Comment by justwagle** [comment]
@hkaiser done.

File: src/execution_tree/primitives/sub_operation.cpp [hunk 3]
```diff
@@ -93,37 +133,16 @@ namespace phylanx { namespace execution_tree { namespace primitives
...
-            array_map_type first(val1.data(), lhs_size);
-            Eigen::Matrix<double, Eigen::Dynamic, 1> result =
-                    std::accumulate(ops.begin()+1, ops.end(),
-                                    array_type{first},
-                                    [&](array_type const& result, ir::node_data<double> const& curr)
-                                            ->  array_type
+        array_type first = ops.begin()->matrix().array();
+        result_type result = std::accumulate(ops.begin()+1, ops.end(),
+                                    first,
+                                    [&](array_type& result, ir::node_data<double> const& curr) ->  array_type
```
Comment (hkaiser) [inline src/execution_tree/primitives/sub_operation.cpp]: Please keep the line length under 90 characters, also using clang-format helps maintaining formatting consistency.

**Pull request merged**